analyzers/src/SonarAnalyzer.Common/Rules/DeclareTypesInNamespacesBase.cs

```diff
@@ -43,8 +42,7 @@ public abstract class DeclareTypesInNamespacesBase<TSyntaxKind> : SonarDiagnosti
               c =>
               {
                   var declaration = c.Node;
-
-                  if (c.ContainingSymbol.Kind != SymbolKind.NamedType
+                  if (c.IsRedundantPositionalRecordContext()
```

Comment:
Optional: There are just 2 conditions that are simple and have explicit name. It can be on a single ine.